app/models/manageiq/providers/vmware/cloud_manager/orchestration_template.rb

```diff
@@ -29,7 +31,64 @@ def vapp_parameters
     ]
   end
 
+  def vm_param_groups
+    # Parse template's OVF file
+    ovf_doc = MiqXml.load(content)
+
+    groups = []
+    # Parse the XML template document for specific vCloud attributes.
+    ovf_doc.root.each_element("//vcloud:GuestCustomizationSection") do |el|
+      vm_id = el.elements["vcloud:VirtualMachineId"].text
+      vm_name = el.elements["vcloud:ComputerName"].text
+
+      groups << OrchestrationTemplate::OrchestrationParameterGroup.new(
+        :label      => vm_name,
+        :parameters => [
+          # Name of the provisioned instance.
+          OrchestrationTemplate::OrchestrationParameter.new(
+            :name          => "instance_name-#{vm_id}",
+            :label         => "Instance name",
+            :data_type     => "string",
+            :default_value => vm_name
+          ),
+
+          # List of available VDC networks.
+          OrchestrationTemplate::OrchestrationParameter.new(
+            :name          => "vdc_network-#{vm_id}",
+            :label         => "Network",
+            :data_type     => "string",
+            :default_value => "(default)",
+            :constraints   => [
+              OrchestrationTemplate::OrchestrationParameterAllowed.new(:allowed_values => vdc_networks)
+            ]
+          )
+        ]
+      )
+    end
+
+    groups
+  end
+
   def self.eligible_manager_types
     [ManageIQ::Providers::Vmware::CloudManager]
   end
+
+  private
+
+  def vdc_networks
+    # In order to be able to deploy this vApp we need to have a valid EMS.
+    ems = ExtManagementSystem.find(ems_id) unless ems_id.nil?
+
+    # Get all VDC networks from the provider
+    vdc_network_type = ManageIQ::Providers::Vmware::NetworkManager::CloudNetwork::OrgVdcNet.name
+    cloud_networks = ems.try(:cloud_networks).try(:where, :type => vdc_network_type)
+
+    # If none exists, ensure user can't select anything (default network will be used)
+    return "(default)" if cloud_networks.empty?
+
+    # Get network names.
+    vdc_network_names = cloud_networks.collect(&:name)
```

Comment:
@gberginc I like the idea but what keys were you thinking of using and what would it make simpler?  I could see `:default => "(default)"` instead of using the string directly.
We'll want to check with @bzwei about changing `OrchestrationParameterAllowed` to support this